db/c.cc

```diff
@@ -1141,6 +1141,18 @@ void rocksdb_column_family_handle_destroy(
   delete handle;
 }
 
+rocksdb_column_family_handle_t* rocksdb_get_default_column_family_handle(
+    rocksdb_t* db) {
+  rocksdb_column_family_handle_t* handle = new rocksdb_column_family_handle_t;
+  handle->rep = db->rep->DefaultColumnFamily();
+  return handle;
+}
+
+void rocksdb_default_column_family_handle_destroy(
+    rocksdb_column_family_handle_t* handle) {
```

Comment:
It's kind of confusing because (1) it has "destroy" in the name but doesn't go through our regular destroy path (`DestroyColumnFamilyHandle()` or `~ColumnFamilyHandle()`); and (2) it has "default" in the name but operates on any `rocksdb_column_family_handle_t*`.

What do you think about simply not offering such an API, and let the user call `delete handle` instead?

Reply:
The reason I add this different api is that, I observed that "delete handle" api will destroy the cf together with the handle, for other cf this behavior is rational because except for the handle, user has no other way to get and work on it, free the handle should also free the cf, too.
But for the default cf, things have a little change, only the new batched multi_get api need the handle arg, what if I finish the call and just want to free the memory of default cf_handle? Maybe another api name is more suitable, what do you think?
Or we can modify the batched multi_get api, if the cf_handle arg is NULL, actions will go to the default cf by default? Maybe adding a new api "rocksdb_batched_multi_get" which use the default cf is more direct, as all other APIs have two version -- with or without "_cf" ending, except for batched multi_get.

Reply:
Oh ok I see the ambiguity of the term "destroy" in our C API is preexisting. How about this as the API change?

```
struct rocksdb_column_family_handle_t {
  ColumnFamilyHandle* rep;
  bool immortal;  // <-- New field
};

extern ROCKSDB_LIBRARY_API rocksdb_column_family_handle_t*
    rocksdb_get_default_column_family_handle(rocksdb_t* db);  // <-- New function
```

- `rocksdb_get_default_column_family_handle()` will set `immortal == true`
- Other creations of `rocksdb_column_family_handle_t`, where `rep` is not obtained by `DefaultColumnFamilyHandle()`, will set `immortal == false`
- `rocksdb_column_family_handle_destroy()` will delete `rep` only when `!immortal`

Reply:
OK, that seems rational, thank you. 
Can I expect to see this new API in the next release?

Reply:
Yes, we will be able to include it in the 9.2 release as long as we land it by Friday this week.

Reply:
Hi @ajkr, 
I notice v9.2.1 release lanched last week, but I can't find the <rocksdb_get_default_column_family_handle> API, is there anything I need to do? I noticed some formart check has failed.

Reply:
I think you still need to update the PR according to the comments in order for us to accept it and land it.

Reply:
hi @ajkr ，
I have modified and make a pull request again, if there is anything I should do (I am fresh man), please let me know, thanks.